src/hooks/useTxDetails/useTxDetails.ts

```diff
@@ -45,22 +45,52 @@ export interface TxDetails {
   sourceMarketData: MarketData
   destinationMarketData: MarketData
   feeMarketData: MarketData
+  i18n: string
+  negated: boolean
 }
 
 export const getStandardTx = (tx: Tx) => (tx.transfers.length === 1 ? tx.transfers[0] : undefined)
-export const getBuyTx = (tx: Tx) => tx.transfers.find(t => t.type === chainAdapters.TxType.Receive)
-export const getSellTx = (tx: Tx) => tx.transfers.find(t => t.type === chainAdapters.TxType.Send)
+export const getTransferByType = (tx: Tx, txType: TxType) =>
+  tx.transfers.find(t => t.type === txType)
+export const getBuyTx = (tx: Tx) => getTransferByType(tx, chainAdapters.TxType.Receive)
+export const getSellTx = (tx: Tx) => getTransferByType(tx, chainAdapters.TxType.Send)
+export const getTransferByAsset = (tx: Tx, asset: Asset) =>
+  tx.transfers.find(t => t.caip19 === asset.caip19)
 
 export const isSupportedContract = (tx: Tx) =>
   tx.data?.method ? SUPPORTED_CONTRACT_METHODS.has(tx.data?.method) : false
 
-export const useTxDetails = (txId: string, activeAsset?: Asset): TxDetails => {
+export const useTxDetails = (
+  txId: string,
+  activeAsset?: Asset,
+  account: string = ''
+): TxDetails => {
   const tx = useAppSelector((state: ReduxState) => selectTxById(state, txId))
   const method = tx.data?.method
 
   const standardTx = getStandardTx(tx)
-  const buyTx = getBuyTx(tx)
-  const sellTx = getSellTx(tx)
+  const buyTx = getTransferByType(tx, chainAdapters.TxType.Receive)
+  const sellTx = { ...getTransferByType(tx, chainAdapters.TxType.Send) } as chainAdapters.TxTransfer
+  const tradeTx = activeAsset?.caip19 === sellTx?.caip19 ? sellTx : buyTx
+  const txMetaData = { ...tx.data } as chainAdapters.TxMetadata
+  const activeTx = activeAsset !== undefined && getTransferByAsset(tx, activeAsset)
+  let i18n = 'unknown'
```

Comment:
As `i18n` is only used in `TransactionContract`, and will likely only ever be, can we move this logic there?